### DISCUSSION TITLE
Listing and Openess of Databases via HTTP API

### DISCUSSION THREAD

**Discussion by gramian** [ArcadeData/arcadedb#598] [Q&A] [answered]
Dear Community,

experimenting with the [HTTP API](https://docs.arcadedb.com/#HTTP-API), I came across two related issues concerning listing databases and testing if databases are open:

1. The `/api/v1/databases` endpoint returns a list of **open** databases, but closed databases are not listed. Is this intentional? The documentation states [accessible databases](https://docs.arcadedb.com/#HTTP-ListDatabases) are returned, but since it could be opened, it is accessible (?). So, if this is intentional, I think the documentation should be extended to state that only **open** databases are listed.

2. What is the best practice to check if a database is **open** via the HTTP API? Currently I am executing a simple SQL query (`SELECT true`) on the database in question, and see if an error is returned or not. Is there a better (as in: not error provoking) way to test if a database is open?

Thank you for your help

**Comment by gramian** [comment]
@lvca Looking through the source, particularly https://github.com/ArcadeData/arcadedb/blob/main/network/src/main/java/com/arcadedb/remote/RemoteDatabase.java#L111 I found an `exists` endpoint; is this a left-over or undocumented? It would be an answer to this question, as it works also for closed databases.

**Comment by lvca** [comment, marked as answer]
Thanks to @gramian to have provided the docs for the `/exists` HTTP command.
